packages/element-lib/src/sidetree/op/index.js

```diff
@@ -117,6 +127,179 @@ const getDeletePayload = async (didUniqueSuffix, recoveryPrivateKey) => {
   return makeSignedOperation(header, payload, recoveryPrivateKey);
 };
 
+const walletToInitionalDIDDoc = wallet => {
+  const didDocumentModel = {
+    '@context': [
+      'https://www.w3.org/ns/did/v1',
+      'https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld',
+    ],
+  };
+
+  const publicKeys = [];
+  const commonVerificationMethods = [];
+  const keyAgreementkeys = [];
+
+  Object.values(wallet.keys).forEach(walletKey => {
+    if (walletKey.type === 'assymetric') {
+      if (walletKey.tags[0] === 'X25519KeyAgreementKey2019') {
+        keyAgreementkeys.push({
+          id: walletKey.tags[1] || `#${walletKey.kid}`,
+          type: 'X25519KeyAgreementKey2019',
+          usage: 'signing',
+          [walletKey.didPublicKeyEncoding]: walletKey.publicKey,
+        });
+      } else {
+        if (walletKey.encoding === 'jwk') {
+          publicKeys.push({
+            id: walletKey.tags[1] || `#${walletKey.kid}`,
+            type: walletKey.tags[0],
+            usage: 'signing',
+            [walletKey.didPublicKeyEncoding]: JSON.parse(walletKey.publicKey),
+          });
+        } else {
+          publicKeys.push({
+            id:
+              walletKey.tags[1] !== undefined
+                ? walletKey.tags[1]
+                : `#${walletKey.kid}`,
+            type: walletKey.tags[0],
+            usage:
+              walletKey.tags[1] &&
+              walletKey.tags[1].split('#').pop() === 'recovery'
+                ? 'recovery'
+                : 'signing',
+            [walletKey.didPublicKeyEncoding]: walletKey.publicKey,
+          });
+        }
+
+        commonVerificationMethods.push(`#${walletKey.kid}`);
+      }
+    }
+  });
+  didDocumentModel.publicKey = publicKeys;
+  didDocumentModel.authentication = commonVerificationMethods;
+  didDocumentModel.assertionMethod = commonVerificationMethods;
+  didDocumentModel.capabilityDelegation = commonVerificationMethods;
+  didDocumentModel.capabilityInvocation = commonVerificationMethods;
+  didDocumentModel.keyAgreement = keyAgreementkeys;
+  return didDocumentModel;
+};
+
+const addDIDToWallet = (did, wallet) => {
+  Object.values(wallet.keys).forEach(walletKey => {
+    if (walletKey.tags[1] === undefined) {
+      walletKey.tags.push(`#${walletKey.kid}`);
+    }
+    const fragment = walletKey.tags[1];
+    walletKey.tags.push(did + fragment);
+  });
+
+  return wallet;
+};
+
+const getNewWallet = async () => {
+  const mnemonic = await MnemonicKeySystem.generateMnemonic();
+  const ed25519Key = await elementCrypto.ed25519.createKeys();
+  const x25519Key = await elementCrypto.ed25519.X25519KeyPair.fromEdKeyPair({
+    publicKeyBase58: ed25519Key.publicKey,
+    privateKeyBase58: ed25519Key.privateKey,
+  });
+
+  const wall = DIDWallet.create();
+
+  const notes = 'generated in element-lib.';
+
+  wall.addKey({
```

Comment:
why add the mnemonic?

Reply:
we use it to generate all keys using the MKS system... if we want to be able to get back to all keys in an indexable manner, we need to add it.

See this issue for more context: https://github.com/w3c-ccg/webkms/issues/2